src/nanorc/node.py

```diff
@@ -507,9 +507,26 @@ def _on_enter_callback(self, event):
                 done = []
                 for child_node in appset:
                     if not child_node.included: continue
-
-                    if not child_node.sup.desc.proc.is_alive() or not child_node.sup.commander.ping():
+                    IsAlive = child_node.sup.desc.proc.is_alive()
+                    if not IsAlive:
+                        failed.append(child_node.name)
+                        self.console.print(f"{child_node.name}, is dead")
+                        child_node.to_error(
+                            command = command,
+                        )
+                        done += [child_node]
+                        break
+                    IsPing = False
+                    pingattempt=5
+                    for i in range(pingattempt):
```

Comment:
This code is part of a loop already (line 500). is a separate nested loop with an internal, non configurable timeout really necessary or couldn't the logic be fitted in the outer loop?

Reply:
We could try but I tried to keep to the same logic as it was in before